Src/FluentAssertions/Specialized/ExecutionTimeAssertions.cs

```diff
@@ -86,6 +85,8 @@ public ExecutionTimeAssertions(ExecutionTime executionTime)
                 elapsed);
 
         return new AndConstraint<ExecutionTimeAssertions>(this);
+
+        bool Condition(TimeSpan duration) => duration <= maxDuration;
```

Comment:
Do you mean inlining into the `.ForCondition()`? But we use it above as well (at `PollUntil()`).

Reply:
What about make it a private method instead of a local function?

Reply:
Inlining will make a better IMO. E.g. `.ForCondition(Condition(elapsed))` doesn't say anything to me. What's the condition here?

Reply:
This local function is used up there too: https://github.com/fluentassertions/fluentassertions/blob/f1a01b1ef98e2b9c76a468b68ef61e196fdb34a3/Src/FluentAssertions/Specialized/ExecutionTimeAssertions.cs#L75-L76

This could be rewritten into: `PollUntil(elapsed => elapsed <= maxDuration, expectedResult: false, rate: maxDuration);`

What do you think?

Reply:
I can imagine this was introduced because of DRY

Reply:
See [22bffcc](https://github.com/fluentassertions/fluentassertions/pull/2264/commits/22bffccd9354985f8e96c5f87afcee26ad889cc1)

Reply:
DRY is not a good argument if it doesn't improve readability. I like your suggestion from 9 minutes ago.

Reply:
Cool have rewritten it :)